Microsoft.Toolkit.Services/Core/DataProviderBase.cs

```diff
@@ -0,0 +1,101 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Services.Exceptions;
+using System.Net.Http;
+
+namespace Microsoft.Toolkit.Services
+{
+    /// <summary>
+    /// Base class for data providers in this library.
+    /// </summary>
+    /// <typeparam name="TConfig">Query configuration type for given provider.</typeparam>
+    public abstract class DataProviderBase<TConfig>
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public DataProviderBase()
+        {
+            HttpClient = httpClient ?? new HttpClient();
```

Comment:
When the BingProvider is instantiated, the HttpClient isn't set and we would instantiate without needing it only to throw it away later